### PR TITLE
Remove filled paths with zero area or stroked paths with zero stroke-width

### LINKED ISSUE #157: remove_unpainted_shapes should drop 0 area unstroked
`<rect x="-1228" y="-57.67" style="fill:#F2F2F2;" width="0" height="0.67"/>` should be dropped by remove_unpainted_shapes; 0 area and no stroke

### PR DESCRIPTION
**Pull request by anthrotype**
Fixes #157

### CODE REVIEW

**Comment by anthrotype** [comment]
ping @rsheeter

File: src/picosvg/svg_pathops.py [hunk 1]
```diff
@@ -190,3 +190,11 @@ def stroke(
 
 def bounding_box(svg_cmds: SVGCommandSeq):
     return skia_path(svg_cmds, fill_rule="nonzero").bounds
+
+
+def path_area(svg_cmds: SVGCommandSeq, fill_rule: str, simplified=True) -> float:
```
Comment (rsheeter) [inline src/picosvg/svg_pathops.py:195]: You kind of have to dig to find out what simplify means. Suggest comment or just no option, always simplify before computing area.

File: src/picosvg/svg_types.py [hunk 2]
```diff
@@ -199,8 +199,12 @@ def _visible(fill, opacity):
         if all(c[0].upper() == "M" for c in self.as_cmd_seq()):
             return False
 
-        return _visible(shape.stroke, shape.stroke_opacity) or _visible(
-            shape.fill, shape.fill_opacity
+        return (
+            _visible(shape.stroke, shape.stroke_opacity) and shape.stroke_width != 0
+        ) or (
+            _visible(shape.fill, shape.fill_opacity)
+            and svg_pathops.path_area(shape.as_cmd_seq(), fill_rule=shape.fill_rule)
```
Comment (rsheeter) [inline src/picosvg/svg_types.py:206]: This is becoming rather ugly. I suggest breaking it up, akin to:

```python
# Does it look like the stroke is visible?
if _visible(shape.stroke, shape.stroke_opacity) and shape.stroke_width != 0:
  return True

# No stroke; if the shape is hidden we can't draw
if not _visible(shape.fill, shape.fill_opacity):
   return False

# Only shapes with area paint
return svg_pathops.path_area(shape.as_cmd_seq(), fill_rule=shape.fill_rule) > 0
```

**Review by rsheeter — APPROVED** [review]

File: src/picosvg/svg_pathops.py (hunk 1 above)
Comment (anthrotype) [reply src/picosvg/svg_pathops.py]: I removed the `simplified` option

File: src/picosvg/svg_types.py (hunk 2 above)
Comment (anthrotype) [reply src/picosvg/svg_types.py]: Done

**Pull request merged**